compiler-core/src/error.rs

```diff
@@ -270,6 +270,9 @@ file_names.iter().map(|x| x.as_str()).join(", "))]
 
     #[error("The Gleam module {path} would overwrite the Erlang module {name}")]
     GleamModuleWouldOverwriteStandardErlangModule { name: EcoString, path: Utf8PathBuf },
+
+    #[error("--replace flag not used to replace a release")]
+    ReplaceRequired,
```

Comment:
Let's call this `HexPublishReplaceRequired` as it's a bit cryptic otherwise, and mention Hex in the message above